arch/xtensa/src/esp32s2/esp32s2_psram.c

```diff
@@ -0,0 +1,729 @@
+#define FLASH_CS_IO           SPI_CS0_GPIO_NUM
+
+/* PSRAM clock and cs IO should be configured based on hardware design. */
+
+#define PSRAM_CLK_IO          CONFIG_ESP32S2_DEFAULT_PSRAM_CLK_IO /* Default value is 30 */
+#define PSRAM_CS_IO           CONFIG_ESP32S2_DEFAULT_PSRAM_CS_IO  /* Default value is 26 */
+#define PSRAM_SPIQ_SD0_IO     SPI_Q_GPIO_NUM
+#define PSRAM_SPID_SD1_IO     SPI_D_GPIO_NUM
+#define PSRAM_SPIWP_SD3_IO    SPI_WP_GPIO_NUM
+#define PSRAM_SPIHD_SD2_IO    SPI_HD_GPIO_NUM
+
+#define CS_PSRAM_SEL          SPI_MEM_CS1_DIS_M
+#define CS_FLASH_SEL          SPI_MEM_CS0_DIS_M
+
+#define PSRAM_IO_MATRIX_DUMMY_20M   0
+#define PSRAM_IO_MATRIX_DUMMY_40M   0
+#define PSRAM_IO_MATRIX_DUMMY_80M   0
+#define _SPI_CACHE_PORT             0
+#define _SPI_FLASH_PORT             1
+#define _SPI_80M_CLK_DIV            1
+#define _SPI_40M_CLK_DIV            2
+#define _SPI_20M_CLK_DIV            4
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+typedef enum
+{
+  PSRAM_CLK_MODE_NORM = 0,  /* Normal SPI mode */
+  PSRAM_CLK_MODE_A1C,       /* ONE extra clock cycles after CS is set high level */
+  PSRAM_CLK_MODE_A2C,       /* Two extra clock cycles after CS is set high level */
+  PSRAM_CLK_MODE_ALON,      /* clock always on */
+  PSRAM_CLK_MODE_MAX,
+} psram_clk_mode_t;
+
+typedef enum
+{
+  PSRAM_EID_SIZE_16MBITS = 0,
+  PSRAM_EID_SIZE_32MBITS = 1,
+  PSRAM_EID_SIZE_64MBITS = 2,
+} psram_eid_size_t;
+
+typedef struct
+{
+  uint8_t flash_clk_io;
+  uint8_t flash_cs_io;
+  uint8_t psram_clk_io;
+  uint8_t psram_cs_io;
+  uint8_t psram_spiq_sd0_io;
+  uint8_t psram_spid_sd1_io;
+  uint8_t psram_spiwp_sd3_io;
+  uint8_t psram_spihd_sd2_io;
+} psram_io_t;
+
+#define PSRAM_IO_CONF_DEFAULT() {             \
+    .flash_clk_io       = FLASH_CLK_IO,       \
+    .flash_cs_io        = FLASH_CS_IO,        \
+    .psram_clk_io       = PSRAM_CLK_IO,       \
+    .psram_cs_io        = PSRAM_CS_IO,        \
+    .psram_spiq_sd0_io  = PSRAM_SPIQ_SD0_IO,  \
+    .psram_spid_sd1_io  = PSRAM_SPID_SD1_IO,  \
+    .psram_spiwp_sd3_io = PSRAM_SPIWP_SD3_IO, \
+    .psram_spihd_sd2_io = PSRAM_SPIHD_SD2_IO,  \
+}
+
+typedef enum
+{
+  PSRAM_SPI_1  = 0x1,
+  PSRAM_SPI_MAX ,
+} psram_spi_num_t;
+
+typedef enum
+{
+  PSRAM_CMD_QPI,
+  PSRAM_CMD_SPI,
+} psram_cmd_mode_e;
+
+uint32_t g_psram_id;
+uint32_t g_psram_size;        /* physical psram size in bytes */
+
+static uint8_t g_psram_cs_io = UINT8_MAX;
+
+extern uint32_t esp_rom_efuse_get_flash_gpio_info(void);
+extern uint32_t esp_rom_efuse_get_flash_wp_gpio(void);
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static void IRAM_ATTR psram_cache_init(int psram_cache_mode,
+                                       int vaddrmode);
```

Comment:
```suggestion
static void IRAM_ATTR psram_cache_init(int psram_cache_mode, int vaddrmode);
```
This seems to fit in a single line.